pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -840,6 +840,19 @@ func (p *provider) newVirtualMachine(c *Config, pc *providerconfig.Config, machi
 			DataVolumeTemplates: getDataVolumeTemplates(c, dataVolumeName, dvAnnotations),
 		},
 	}
+
+	if isDedicatedVCPURequested(machine.Spec.Annotations) {
+		cpusAsUint64, err := strconv.ParseUint(c.CPUs, 0, 64)
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse cpu cores: %w", err)
+		}
+
+		virtualMachine.Spec.Template.Spec.Domain.CPU = &kubevirtcorev1.CPU{
+			Cores: uint32(cpusAsUint64),
+		}
+	}
```

Comment:
If this annotation is consumed in machine-controller then this should be configured at `machine.Annotations` not in the spec.